pkgs/build-support/fetchgithub/default.nix

```diff
@@ -55,7 +57,13 @@ let
       inherit rev deepClone fetchSubmodules sparseCheckout; url = gitRepoUrl;
     } // lib.optionalAttrs (leaveDotGit != null) { inherit leaveDotGit; }
     else {
-      url = "${baseUrl}/archive/${rev}.tar.gz";
```

Comment:
This means the api endpoint would be always used, even if no token is provided. Correct?
Would this not mean that public CI that does not access private repositories and therefore has no token would run into API limits?

Reply:
Correct, although I would be very surprised if there is a different rate limit applied to this new URL than to the previous one.

Reply:
The other one does not have rate limits. This would also break builds if many people are behind a shared ip address. i.e. this would break in github action runner.